test/system/260-sdnotify.bats

```diff
@@ -371,4 +371,100 @@ READY=1" "sdnotify sent MAINPID and READY"
     run_podman rmi $(pause_image)
 }
 
+# The functione takes the following arguments:
+# 1. File to write to
+# 2. Command container 1
+# 3. Command container 2
+# 4. Sdnotify policy
+function generate_exit_code_yaml {
+    echo "
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    app: test
+  name: test_pod
+  annotations:
+    io.containers.sdnotify: "$4"
+spec:
+  restartPolicy: Never
+  containers:
+    - name: ctr1
+      image: $IMAGE
+      command:
+      - $2
+    - name: ctr2
+      image: $IMAGE
+      command:
+      - $3
+" > $1
+}
+
+@test "podman kube play - exit-code policy" {
+    fname="$PODMAN_TMPDIR/test.yaml"
+
+    # Create a test matrix with the following arguments:
+    # 1. Exit policy
+    # 2. Command of container 1
+    # 3. Command of container 2
+    # 4. Expected exit code of the service container
+    exit_tests="
+all  | true  | true  | 0
+all  | true  | false | 0
+all  | false | false | 137
+any  | true  | true  | 0
+any  | false | true  | 137
+any  | false | false | 137
+none | true  | true  | 0
+none | true  | false | 0
+none | false | false | 0
+"
+
+    # I am sorry, this is a long test as we need to test the upper matrix
+    # twice. The first run is using the default sdnotify policy of "ignore".
+    # In this case, the service container serves as the main PID of the service
+    # to have a minimal resource footprint.  The second run is using the
+    # "conmon" sdnotify policy in which case Podman needs to serve as the main
+    # PID to act as an sdnotify proxy; there Podman will wait for the service
+    # container to exit and reflects its exit code.
+    while read exit_policy cmd1 cmd2 exit_code; do
+        for sdnotify_policy in ignore conmon; do
+            generate_exit_code_yaml $fname $cmd1 $cmd2 $sdnotify_policy
+            yaml_sha=$(sha256sum $fname)
+            service_container="${yaml_sha:0:12}-service"
+            podman_exit=0
+            if [[ $sdnotify_policy != "ignore" ]];then
+                 podman_exit=$exit_code
+            fi
```

Comment:
My stupid old brain again. It just doesn't like negative tests. I might find this more readable
```bash
    podman_exit=$exit_code
    if [[ $sdnotify_policy == "ignore" ]]; then
        podman_exit=0
    fi
```